droidlet/lowlevel/hello_robot/hello_robot_mover.py

```diff
@@ -56,30 +56,74 @@ def __init__(self, ip=None):
         self.bot = Pyro4.Proxy("PYRONAME:hello_robot@" + ip)
         self.bot._pyroAsync()
         self.is_moving = self.bot.is_moving()
+        self.camera_transform = self.bot.get_camera_transform().value
+        self.camera_height = self.camera_transform[2, 3]
         self.cam = Pyro4.Proxy("PYRONAME:hello_realsense@" + ip)
 
         self.data_logger = Pyro4.Proxy("PYRONAME:hello_data_logger@" + ip)
         self.data_logger._pyroAsync()
         _ = safe_call(self.data_logger.ready)
         self.curr_look_dir = np.array([0, 0, 1])  # initial look dir is along the z-axis
 
-        intrinsic_mat = np.asarray(safe_call(self.cam.get_intrinsics))
+        intrinsic_mat = safe_call(self.cam.get_intrinsics)
         intrinsic_mat_inv = np.linalg.inv(intrinsic_mat)
-        img_resolution = safe_call(self.cam.get_img_resolution)
-        img_pixs = np.mgrid[0 : img_resolution[1] : 1, 0 : img_resolution[0] : 1]
+        img_resolution = safe_call(self.cam.get_img_resolution, rotate=False)
+        img_pixs = np.mgrid[0 : img_resolution[0] : 1, 0 : img_resolution[1] : 1]
         img_pixs = img_pixs.reshape(2, -1)
         img_pixs[[0, 1], :] = img_pixs[[1, 0], :]
         uv_one = np.concatenate((img_pixs, np.ones((1, img_pixs.shape[1]))))
         self.uv_one_in_cam = np.dot(intrinsic_mat_inv, uv_one)
 
-    def log_data(self, seconds):
+    def log_data_start(self, seconds):
         self.data_logger.save_batch(seconds)
 
+    def log_data_stop(self):
+        self.data_logger.stop()
+
     def bot_step(self):
-        pass
+        f = not self.bot.is_moving().value
+        return f
+
+    def look_at(self, target, turn_base=True):
+        """
+        Executes "look at" by setting the pan, tilt of the camera
+        or turning the base if required.
+        Uses both the base state and object coordinates in 
+        canonical world coordinates to calculate expected yaw and pitch.
+
+        Args:
+            target (list): object coordinates as saved in memory.
+            turn_base: if False, will try to look at point only by moving camera and not base
+        """
+        old_pan = self.get_pan()
+        old_tilt = self.get_tilt()
+        pos = self.get_base_pos_in_canonical_coords()
+        cam_transform = self.bot.get_camera_transform().value
+        cam_pos = cam_transform[0:3, 3]
+        # convert cam_pos to canonical co-ordinates
+        cam_pos = [pos[0], cam_pos[2], pos[1]]
+
+        logging.info(f"Current base state (x, z, yaw): {pos}, camera state (x, y, z): {cam_pos}")
+        logging.info(f"looking at x,y,z: {target}")
+        
+        pan_rad, tilt_rad = get_camera_angles(cam_pos, target)
+        tilt_rad = -tilt_rad
```

Comment:
please add a comment for this sign inversion. why is this needed?

Reply:
added a comment